cleanup old not functioning epel and yumrepo variables from the redhat o...

abenari opened the pull request:
...s family code

replacing it with a place holder for repos, that can be override by a plugin.

domcleal commented on app/models/operatingsystems/redhat.rb:
```diff
@@ -17,29 +17,9 @@ def mediumpath host
-      return ""
-    end
-    "su -c 'rpm -Uvh #{medium_uri(host, epel_url)}'"
+  def repos host
```
The RedHat family seems the wrong place for it, is it not more universal than that?

domcleal commented:
The interface for 'repos' seems ill-defined to me, can we state what sort of data is expected or provide more structure around it than just expecting a plugin to override it?

A redmine ticket is also required explaining what's being changed + why, thanks.

abenari commented:
@domcleal I see your point about the interface definition, how do you suggest defining the interface?

domcleal commented:
@abenari at the very simplest a comment so we know whether it's passing back an array of hashes or whatever, but I think we should probably look to something like YARD to begin documenting extension points that we expect people to override or not (e.g. `@api public`).

How are you overriding it at the moment in the plugin?  By changing the repos method in the OS model when your plugin's loaded?

abenari commented:
@domcleal added documentation in the method and renamed the commit to point to the the redmine issue.

domcleal commented:
@abenari that looks good, thanks.

I see `epel` in Operatingsystem::Jail (app/models/operatingsystem.rb:54), I guess this needs changing to `repos` so you're able to access this helper in safemode?

abenari committed:
fixes #2837 add an interface to add repositories to kickstart

also clean up old methods that where ment to do the same but doesn't
work anymore.

domcleal commented:
Merged as 6b27556 with a typo fix, thanks @abenari.
